java/CMakeLists.txt

```diff
@@ -309,13 +309,17 @@ set(JAVA_TESTCLASSPATH ${JAVA_JUNIT_JAR} ${JAVA_HAMCR_JAR} ${JAVA_MOCKITO_JAR} $
 set(JNI_OUTPUT_DIR ${PROJECT_SOURCE_DIR}/java/include)
 file(MAKE_DIRECTORY ${JNI_OUTPUT_DIR})
 
+if(${Java_VERSION_MINOR} VERSION_LESS_EQUAL "7" AND ${Java_VERSION_MAJOR} STREQUAL "1")
```

Comment:
@alanpaxton Just wondering, as they have changed the version numbering scheme after Java 1.8, does this still work on say Java 11? How about Java 17 ;-)

Reply:
The values come from CMake's FindJava plugin. I just went and played with it. Java 11, 13, 15, 17 all have correct ${Java_VERSION_MAJOR} of 11,. 13, 15, 17 respectively. Java 1.6, 1.7 and 1.8 have ${Java_VERSION_MAJOR} STREQUAL "1" so I claim this test works as intended.